### PR TITLE
#163388243 Configure Authors Haven to use PostgreSQL as DB Engine

### PR DESCRIPTION
**Pull request by bryan-munene**
**What does this PR do?**

SetsUp Postgresql Database as the default database for the application

**Description of Task to be completed?**

Download and configure postgresql database to be the default database for the application

**How should this be manually tested?**

Clone the repo and cd into it

. checkout to the branch git checkout Ch-set-up-database-163388243

. pip install -r requirements.txt

Install postgresql database on your PC.

- create database 'authorshaven'

- create user 'birdbox'

- grant all privileges to database 'authorshaven' to user 'birdbox'

- set the user password as a OS enviroment variable on the terminal. `export DBpass="your password" `

- `python3 manage.py makemigrations` then `python3 manage.py migrate` to test the database 

**Any background context you want to provide?**

None

**What are the relevant pivotal tracker stories?**

[#163388238](https://www.pivotaltracker.com/story/show/163388243)

**Screenshots (if appropriate)**

None

### CODE REVIEW

File: authors/settings.py [hunk 1]
```diff
@@ -81,8 +81,12 @@
 
 DATABASES = {
     'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
+        'ENGINE': 'django.db.backends.postgresql',
```
Comment (sam-karis) [inline authors/settings.py]: Hi remove the hard-coded parameters, I suggest you pick them from the environment and you can also try to condense the database url to one variable e.g. `postgresql://username@localhost:5432/db_name`

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,12 @@
-Django==1.10
-django-cors-middleware==1.2.0
-django-extensions==1.7.1
-djangorestframework==3.4.4
-PyJWT==1.4.2
-six==1.10.0
+asn1crypto==0.24.0
```
Comment (sam-karis) [inline requirements.txt:1]: I don't think your task scope required you to edit all these packages.

**Review by sam-karis — CHANGES_REQUESTED** [review]
I suggest you only rebase from `develop` branch only to ensure your PR is easy to understand and follow

File: authors/settings.py (hunk 1 above)
Comment (bryan-munene) [reply authors/settings.py]: Thanks for the feedback. I have implemented it as requested. Please review and let me know


> Hi remove the hard-coded parameters, I suggest you pick them from the environment and you can also try to condense the database url to one variable e.g. `postgresql://username@localhost:5432/db_name`

**Review by sam-karis — CHANGES_REQUESTED** [review]
Hi good work ensure you add the necessary packages e.g(psycopg2, python-decouple,  dj-database-url) in the `requirements.txt`

**Comment by bryan-munene** [comment]
> Hi good work ensure you add the necessary packages e.g(psycopg2, python-decouple, dj-database-url) in the `requirements.txt`

all done.

**Review by sam-karis — APPROVED** [review]
LGTM

**Pull request merged**